linkerd/proxy/http/src/glue.rs

```diff
@@ -274,11 +274,8 @@ where
     }
 }
 
-impl<C> hyper_connect::Connection for Connection<C>
-where
-    C: hyper_connect::Connection,
-{
+impl<C> hyper_connect::Connection for Connection<C> {
     fn connected(&self) -> hyper_connect::Connected {
-        self.transport.connected().proxy(self.absolute_form)
+        hyper_connect::Connected::new().proxy(self.absolute_form)
```

Comment:
What's the deal with `self.transport` now? Unused?

Reply:
`self.transport` is the underlying transport. When I first wrote the `Connection` impl, I bounded the type of the transport with `C: hyper::client::connect::Connected`, which is implemented for `TcpStream`s. However, in practice, it's a `BoxedIo`, and we don't currently forward that trait.

We may want to revisit this in the future and start forwarding the `Connected` metadata through the whole stack, but for now I think this is fine.

Reply:
Oh, I see... this is just a metadata type and not an io stream.

Reply:
Yeah, that's right. Setting `proxy` on the metadata determines whether absolute URIs are used (should probably add a comment explaining that).

There are additional things we can set on the metadata such as whether H2 has already been negotiated and extensions to automatically add on every response received on that connection. We may wish to use these in the future. I believe that when returned by a `TcpStream`, the metadata is empty, so I don't think it's a problem that we don't forward it.